KeeperSdk/vault/ApplyMembership.cs

```diff
@@ -0,0 +1,241 @@
+                { 
+                    (key: p.Uid, perm: p), 
+                    (key: p.Name?.ToLower(), perm: p) 
+                })
+                .Where(x => !string.IsNullOrEmpty(x.key))
+                .GroupBy(x => x.key)
+                .ToDictionary(g => g.Key, g => g.First().perm);
+        }
+
+        private static (string userId, UserType userType) ResolveUserOrTeam(
+            ImportSharedFolderPermissions permission,
+            Dictionary<string, string> teamLookup)
+        {
+            if (!string.IsNullOrEmpty(permission.Uid) && teamLookup.TryGetValue(permission.Uid, out var teamUid))
+                return (teamUid, UserType.Team);
+
+            if (string.IsNullOrEmpty(permission.Name))
+                return (null, UserType.User);
+
+            var name = permission.Name.ToLower();
+            
+            if (teamLookup.TryGetValue(name, out var teamByName))
+                return (teamByName, UserType.Team);
+
+            return IsValidEmail(name) ? (name, UserType.User) : (null, UserType.User);
+        }
+
+        private static bool IsValidEmail(string email)
+        {
+            try { _ = new System.Net.Mail.MailAddress(email); return true; }
+            catch { return false; }
+        }
+
+        private static void TrackProcessedIds(HashSet<string> processedIds, string userId, string name)
+        {
+            processedIds.Add(userId);
+            if (!string.IsNullOrEmpty(name)) processedIds.Add(name.ToLower());
+        }
+
+        private static SharedFolderPermission FindExistingPermission(
+            Dictionary<string, SharedFolderPermission> currentPermissions,
+            string userId,
+            string name)
+            => currentPermissions.TryGetValue(userId, out var byUid) ? byUid
+             : !string.IsNullOrEmpty(name) && currentPermissions.TryGetValue(name.ToLower(), out var byName) ? byName
+             : null;
+
+        private static async Task ApplyPermission(
+            VaultOnline vault,
+            string sharedFolderUid,
+            string userId,
+            UserType userType,
+            SharedFolderUserOptions options,
+            SharedFolderPermission existing,
+            ImportSharedFolderPermissions permission,
+            MembershipSummary summary)
+        {
+            var isUpdate = existing != null;
+            var needsChange = !isUpdate || 
+                              existing.ManageUsers != (permission.ManageUsers ?? false) ||
+                              existing.ManageRecords != (permission.ManageRecords ?? false);
+
+            if (!needsChange)
+                return;
+
+            await vault.PutUserToSharedFolder(sharedFolderUid, userId, userType, options);
+            IncrementSummary(summary, userType, isUpdate);
+        }
+
+        private static void IncrementSummary(MembershipSummary summary, UserType userType, bool isUpdate)
+        {
+            if (isUpdate)
```

Comment:
```suggestion
           if (isUpdate)
              if (userType == UserType.Team) summary.TeamsUpdated++; else summary.UsersUpdated++;
          else
              if (userType == UserType.Team) summary.TeamsAdded++; else summary.UsersAdded++;
```

something along these lines will make it clear